src/Console/Application.php

```diff
@@ -124,13 +109,32 @@ private function isInitialTestPhpOptionHasXdebug(InputInterface $input): bool
         );
     }
 
-    public function doRun(InputInterface $input, OutputInterface $output)
+    protected function doRunCommand(Command $command, InputInterface $input, OutputInterface $output)
     {
+        if ($command->getName() === InfectionCommand::COMMAND_NAME) {
```

Comment:
Clearly this is what an InfectionCommand *should* do. It it their responsibility to check that we're running with xdebug or not. Not Application's.